### ISSUE TITLE
Pyreadline and color code issue in windows 

### ISSUE THREAD

**Issue by jsarver** [xonsh/xonsh#398]
First off, this tool is really awesome and I love it. That being said I am having an unusual issue and I think I've narrowed it down to pyreadline. 

I am running windows 7 Miniconda3 python 3.4.3 64 bit and here are the steps I do to recreate the issue.

```
conda create -n shelltest -c scopatz xonsh
```

It then asks me to install the follwoing which I say yes.

```
The following NEW packages will be INSTALLED:

pip:        7.1.2-py34_0
ply:        3.7-py34_0
python:     3.4.3-0
setuptools: 18.3.2-py34_0
wheel:      0.26.0-py34_1
xonsh:      0.2.0-py34_1
```

Everything installs as expected so I run "Activate shelltest" to activate my virtualenv and type "xonsh" to run my xonsh shell and get the following prompt:

```
☺←[1;32m☻jsarver@mycomputername☺←[1;34m☻ ~☺←[1;32m☻ ☺←[1;34m☻$☺←[0m☻
```

Soo.. I get the odd color codes instead of the default green and blue prompt, that is strange. After some fooling around I discoverd if I install pyreadline then run xonsh I get this prompt:

```
jsarver@mycomputername ~ $
```

Ok now things are looking up, but now if I try to utilize any multiline functionality like defining a function in the xonsh command line like so:

```
def foo():
```

I get the following error.

```
jsarver@mycomputername ~ $ def foo():
pre_input_hook failed
Readline internal error
Traceback (most recent call last):
  File "C:\Miniconda3\envs\shelltest\lib\site-packages\pyreadline\modes\basemode.py", line 124, in readline_setup
    self.pre_input_hook()
  File "C:\Miniconda3\envs\shelltest\lib\site-packages\xonsh\readline_shell.py", line 81, in inserter
    readline.redisplay()
AttributeError: 'module' object has no attribute 'redisplay'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Miniconda3\envs\shelltest\lib\site-packages\pyreadline\console\console.py", line 768, in hook_wrapper_23
    res = ensure_str(readline_hook(prompt))
  File "C:\Miniconda3\envs\shelltest\lib\site-packages\pyreadline\rlmain.py", line 569, in readline
    self.readline_setup(prompt)
  File "C:\Miniconda3\envs\shelltest\lib\site-packages\pyreadline\rlmain.py", line 564, in readline_setup
    BaseReadline.readline_setup(self, prompt)
  File "C:\Miniconda3\envs\shelltest\lib\site-packages\pyreadline\rlmain.py", line 253, in readline_setup
    return self.mode.readline_setup(prompt)
  File "C:\Miniconda3\envs\shelltest\lib\site-packages\pyreadline\modes\basemode.py", line 127, in readline_setup
    traceback.print_exc()
NameError: name 'traceback' is not defined
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK=True
  File "<string>", line None
SyntaxError: None: no further code
```

If I uninstall pyreadline and go back to to funky looking prompt I can define functions all day and it works fine. I'm sure I can just change my prompt to not use color but feel I should report this because it doesn't seem to be working as intended. I did uninstall my python installations and reinstalled a fresh miniconda in the hopes it was just a borked python install, still no luck.  Please let me know if there is anything else I can provide to help troubleshoot this.

Thanks
Josh

**Comment by scopatz** [comment]
Hi @jsarver, thanks for the very detailed report! My first question is have you tried using prompt-toolkit?  See the `$SHELL_TYPE` docs on [this page](http://xonsh.org/envvars.html).  I think that we have been telling Windows users to use prompt-toolkit because of readline and pyreadline issues.

Note that if you want to get rid of the wacky color issues using regular readline, you can always set up a custom `$PROMPT` variable, see [here for more detail](http://xonsh.org/tutorial.html#customizing-the-prompt).  

I am confident that we can work through these issues one way or another.

**Comment by jsarver** [comment]
Hi @scopatz, initially I installed prompt-toolkit and it didn't seem to do anything but then after some further digging in the issue tracker I realized I didn't have it enabled in my xonshrc. Prompt-toolkit definitely appears to have remedied the issue and looks like a superior alternative altogether.

It might be beneficial to us folks stuck with windows to put some sort of recommendation in the install documentation to use prompt-toolkit instead of pyreadline. It's possible I may have missed that recommendation somewhere.

I am happy to use prompt-toolkit for now but if there's any other things you'd like me to try on my end to track down the issue let me know. 

Thanks again,
Josh

**Comment by scopatz** [comment]
> It might be beneficial to us folks stuck with windows to put some sort of recommendation in the install documentation to use prompt-toolkit instead of pyreadline. It's possible I may have missed that recommendation somewhere.

I think that the current docs recommend pyreadline actually.  I would *really* appreciate it if you opened up a PR that fixed the docs in this way. Thanks!

> I am happy to use prompt-toolkit for now but if there's any other things you'd like me to try on my end to track down the issue let me know. 

Great!

**Comment by jsarver** [comment]
As soon as I get a chance I'll update the docs.

Thanks again.

**Comment by scopatz** [comment]
Thanks @jsarver!

**Comment by scopatz** [comment]
There is now a potential fix for this in #2208

**Comment by scopatz** [comment]
Closing due to merge if #2208, please reopen if this persists.
